openapi2kong/oas3_testfiles/18-request-validator-plugin-path-params-outside-ops.yaml

```diff
@@ -0,0 +1,36 @@
+# When the request-validator is added without a body or parameter schema
+# the generator should automatically generate it.
+
+openapi: 3.0.2
+
+info:
+  title: Example
+  version: 1.0.0
+
+servers:
+  - url: http://backend.com/path
+
+x-kong-plugin-request-validator: {}
+
+paths:
+ /test/common-param/{common-param}:
+      parameters:
+        - in: path
+          name: common-param
+          schema:
+            type: integer
+          required: true
+      get:
+        parameters:
+          - in: query
+            name: metadata
+            schema:
+              type: boolean
+            required: false
```

Comment:
can we add a second path-parameter, that gets added to the path-level parameters array, but ALSO to the operation level? The one on the operation level should override the path-level one by the same name.

probably best to also add that description to the top comments explaining the test.

Reply:
Added a different test for this.
Also, for a config like below:

```
paths:
 /test/common-param/{common-param}:
      parameters:
        - in: path
          name: common-param
          schema:
            type: integer
          required: true
      get:
        parameters:
          - in: path
            name: common-param
            schema:
              type: number
            required: true
```
I have added type as number in operation param schema. The request-validator doesn't seem to override by default, as tested. So, if this is the expected case, I have added the logic to skip path-level parameters in case operation one shares the same name and location.